src/mod/applications/mod_http_cache/prometheus_metrics.cpp

```diff
@@ -0,0 +1,136 @@
+#include <map>
+#include <string>
+#include "prometheus_metrics.h"
+
+#define MAX_BUCKET_LEN 8
+#define BUCKET_BOUND 0
+#define BUCKET_COUNT 1
+#define BUCKET_INFINITY UINT_MAX	// not perfect, but good enough
+
+class prometheus_metrics
+{
+public:
+	class auto_lock
+	{
+	public:
+		auto_lock(switch_mutex_t* mutex) : _mutex(mutex)
+		{
+			switch_mutex_lock(_mutex);
+		}
+		~auto_lock()
+		{
+			switch_mutex_unlock(_mutex);
+		}
+		switch_mutex_t* _mutex;
+	};
+	
+	prometheus_metrics(switch_loadable_module_interface_t **module_interface, switch_memory_pool_t* pool) :
+		_pool(pool),
+		_module_interface(module_interface),
+		_download_fail_count(0),
+		_download_buckets{
+			{300, 500, 800, 1300, 2100, 3400, 5500, BUCKET_INFINITY},
+			{0,   0,   0,   0,    0,    0,    0,    0}
+		},
+		_download_bucket_sum(0),
+		_download_bucket_count(0)
+	{
+		switch_mutex_init(&_mutex, SWITCH_MUTEX_NESTED, _pool);
+	}
+	
+	~prometheus_metrics()
+	{
+		switch_mutex_destroy(_mutex);
+	}
+
+	void increment_download_duration(unsigned int duration)
+	{
+		auto_lock lock(_mutex);
+		int i;
+
+		_download_bucket_sum += duration;
+		_download_bucket_count++;
+
+		for (i = 0; i < MAX_BUCKET_LEN; i++) {
```

Comment:
You are in c++ so you can just declare the variable inside `for loop`

Reply:
Indeed, it's a wonderful world. ;)